### PR TITLE
Use predefined padding

### PR DESCRIPTION
**Pull request by karakusc**
Always pads images to one of the predefined set of inputs shapes. Convergence:

^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(bbox)/IoU=0.5: 0.59371
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(bbox)/IoU=0.5:0.95: 0.37408
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(bbox)/IoU=0.75: 0.40668
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(bbox)/large: 0.47926
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(bbox)/medium: 0.40997
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(bbox)/small: 0.22352
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(segm)/IoU=0.5: 0.56112
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(segm)/IoU=0.5:0.95: 0.34377
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(segm)/IoU=0.75: 0.3631
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(segm)/large: 0.46247
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(segm)/medium: 0.37764
^[[32m[0419 12:05:44 @monitor.py:467]^[[0m mAP(segm)/small: 0.18688

### CODE REVIEW

File: MaskRCNN/data.py [hunk 1]
```diff
@@ -24,8 +24,56 @@
 import math
 
 # import tensorpack.utils.viz as tpviz
+padding_shapes = [(800, 1000), (800, 1100), (800, 1200), (800, 1300), (800, 1350)]
```
Comment (armandmcqueen) [inline MaskRCNN/data.py]: Let's move this to a config value

Comment (karakusc) [reply MaskRCNN/data.py]: Done

File: MaskRCNN/config.py [hunk 2]
```diff
@@ -139,6 +139,8 @@ def __ne__(self, _):
 # Un-scaled version: [0.485, 0.456, 0.406], [0.229, 0.224, 0.225]
 _C.PREPROC.PIXEL_MEAN = [123.675, 116.28, 103.53]
 _C.PREPROC.PIXEL_STD = [58.395, 57.12, 57.375]
+_C.PREPROC.PREDEFINED_PADDING = True
+_C.PREPROC.PADDING_SHAPES = [(800, 1000), (800, 1100), (800, 1200), (800, 1300), (800, 1350)]
```
Comment (armandmcqueen) [inline MaskRCNN/config.py]: Add a comment here that while you should only pass in the landscape shapes, we will still create tensors with portrait shape

**Comment by armandmcqueen** [comment]
Looks good. Please add that one clarifying comment and then we can merge it.

File: MaskRCNN/config.py (hunk 2 above)
Comment (karakusc) [reply MaskRCNN/config.py]: Added. Also noted that these should be listed in decreasing h/w order. We can remove this requirement when we revise the algorithm.

**Pull request merged**